frontend/src/components/CreateCollectionModal/components/Content/components/LinkInput/index.tsx

```diff
@@ -0,0 +1,110 @@
+import { IconNames } from "@blueprintjs/icons";
+import { debounce } from "lodash-es";
+import React, { FC } from "react";
+import Input from "src/components/common/Form/Input";
+import { GRAY } from "src/components/common/theme";
+import { DEBOUNCE_TIME_MS } from "src/components/CreateCollectionModal/components/Content/common/constants";
+import { IconWrapper, StyledButton, Wrapper } from "./style";
+
+export type LinkValue = {
+  id: number;
+  value: string;
+  isValid: boolean;
+  index: number;
+};
+
+export enum TYPES {
+  DOI = "DOI",
+  RAW_DATA = "RAW_DATA",
+  PROTOCOL = "PROTOCOL",
+  LAB_WEBSITE = "LAB_WEBSITE",
+  OTHER = "OTHER",
+}
+
+export const OPTIONS = {
+  [TYPES.DOI]: { text: "DOI", value: TYPES.DOI },
+  [TYPES.RAW_DATA]: { text: "Raw Data", value: TYPES.RAW_DATA },
+  [TYPES.PROTOCOL]: { text: "Protocol", value: TYPES.PROTOCOL },
+  [TYPES.LAB_WEBSITE]: { text: "Lab Website", value: TYPES.LAB_WEBSITE },
+  [TYPES.OTHER]: { text: "Other", value: TYPES.OTHER },
+};
+
+const DOI_PLACEHOLDER = "https://doi.org/10.1126/science.aax6234";
+const LINK_PLACEHOLDER = "https://cellxgene.cziscience.com";
+
+interface Props {
+  handleChange: ({ id, value, isValid }: LinkValue) => void;
+  id: number;
+  index: number;
+  type: TYPES;
+  handleDelete: (id: number) => void;
+}
+
+const LinkInput: FC<Props> = ({
+  handleChange,
+  handleDelete,
+  id,
+  type,
+  index,
+}) => {
+  const option = OPTIONS[type];
+
+  const { text, value } = option;
+
+  return (
+    <Wrapper>
+      <Input
+        name={value}
+        text={text}
+        handleChange={debounce(handleChange_, DEBOUNCE_TIME_MS)}
+        syncValidation={[isValidHttpUrl, isDOILink(value)]}
+        noNameAttr
+        placeholder={value === TYPES.DOI ? DOI_PLACEHOLDER : LINK_PLACEHOLDER}
+      />
+      <IconWrapper>
+        <StyledButton
+          minimal
+          color={GRAY.A}
+          icon={IconNames.CROSS}
+          onClick={() => handleDelete(id)}
+        />
+      </IconWrapper>
+    </Wrapper>
+  );
+
+  function handleChange_({
+    isValid: isValidFromInput,
+    value,
+  }: {
+    isValid: boolean;
+    value: string;
+  }) {
+    handleChange({ id, index, isValid: isValidFromInput, value });
+  }
+};
+
+function isDOILink(type: TYPES): (value: string) => true | string {
+  return (value: string) => {
+    if (type !== TYPES.DOI) return true;
```

Comment:
We're returning true on non-doi links because we're checking validity not presence right? Maybe the function name can be a bit clearer

Reply:
Ooh great question! So Checking for `isValidHttpUrl()` is a separate sync validation function that we pass to the `<Input />` component below, so I was thinking `if (type !== TYPES.DOI) return true;` means skipping validation if the type is not `DOI`. Maybe I should add a comment there? E.g.,

```ts
// Skip validation if type is not DOI
if (type !== TYPES.DOI) return true;
```

```tsx
<Input
        name={value}
        text={text}
        handleChange={debounce(handleChange_, DEBOUNCE_TIME_MS)}
        syncValidation={[isValidHttpUrl, isDOILink(value)]} <-- HERE
        noNameAttr
        placeholder={value === TYPES.DOI ? DOI_PLACEHOLDER : LINK_PLACEHOLDER}
      />
```

Reply:
I think the comment would work 👍

Reply:
Awesome thank you! Done!